mangaki/mangaki/forms.py

```diff
@@ -9,6 +10,15 @@ class Meta:
         fields = ['work', 'problem', 'message']
         widgets = {'work': forms.HiddenInput()}
 
+    def __init__(self, *args, **kwargs):
+        work = kwargs.pop('work', None)
+        super(SuggestionForm, self).__init__(*args, **kwargs)
+
+        if work is not None:
+            nsfw_state = 'nsfw' if work.nsfw else 'n_nsfw'
+            new_choices = filter(lambda x: x[0] != nsfw_state, SUGGESTION_PROBLEM_CHOICES)
+            self.fields['problem'].choices = new_choices
```

Comment:
I don't know if it's alright for `choices` to be an iterable-like? Did you test it? (I'm gonna do it anyway in few seconds.)